game-service/src/main/java/org/libertybikes/game/core/GameBoard.java

```diff
@@ -10,61 +10,96 @@
 public class GameBoard {
 
     public static final int BOARD_SIZE = 121;
-    public static final boolean SPOT_AVAILABLE = true, SPOT_TAKEN = false;
+    public static final short SPOT_AVAILABLE = 0, PLAYER_SPOT_TAKEN = 1, OBJECT_SPOT_TAKEN = -1;
 
     // @JsonbTransient // TODO use annotation here once OpenLiberty upgrades to yasson 1.0.1 (contains bug fix)
-    private final boolean[][] board = new boolean[BOARD_SIZE][BOARD_SIZE];
+    private final short[][] board = new short[BOARD_SIZE][BOARD_SIZE];
 
     public final Set<Obstacle> obstacles = new HashSet<>();
+    public final Set<MovingObstacle> movingObstacles = new HashSet<>();
+
     public final Set<Player> players = new HashSet<>();
 
     public GameBoard() {
         for (int i = 0; i < BOARD_SIZE; i++)
             Arrays.fill(board[i], SPOT_AVAILABLE);
     }
 
-    public boolean addObstacle(Obstacle o) {
-        if (o.x + o.width > BOARD_SIZE || o.y + o.height > BOARD_SIZE)
+    public boolean verifyObstacle(Obstacle o) {
```

Comment:
is verifyObstacle called anywhere besides `addObstacle`?  We could just collapse these methods otherwise.

Reply:
There's 2 addObstacle methods that subtly stick the obstacle in a different set

Reply:
this could also be done inside of `verifyObstacle` with an `instanceof` check